### ISSUE TITLE
CPU training epoch exceeds 60 seconds with SVM-silu-norm-qi(4).nf-svm-exp-norm-iq(1).m model on temporal dataset

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#501]
Training epoch exceeded 60-second runtime threshold on CPU backend. Model contains two SVM operations with Silu and Exp activations, batch normalization, and quantized layers (qi(4).nf and iq(1).m) trained on data/temporal/series_sqlite.sqlite with split(0.8). The public Recipe run remained active past the configured slow-runtime limit, indicating the CPU implementation does not achieve proportionate throughput for this model's operation count, data volume, and memory traffic.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `cfe34bfecee92f87`
- Classifier provider: `opencode`
- Classifier model: `nemotron-3.5-lightning-free`
- Classifier effort: `default`

### Classifier decision

```json
{
  "verdict": "new",
  "issue": 0,
  "title": "CPU training epoch exceeds 60 seconds with SVM-silu-norm-qi(4).nf-svm-exp-norm-iq(1).m model on temporal dataset",
  "body": "Training epoch exceeded 60-second runtime threshold on CPU backend. Model contains two SVM operations with Silu and Exp activations, batch normalization, and quantized layers (qi(4).nf and iq(1).m) trained on data/temporal/series_sqlite.sqlite with split(0.8). The public Recipe run remained active past the configured slow-runtime limit, indicating the CPU implementation does not achieve proportionate throughput for this model's operation count, data volume, and memory traffic.",
  "rationale": "No existing GitHub issues were found matching this earliest failing operation combination (SVM + Silu + BatchNorm + qi(4).nf + SVM + Exp + BatchNorm + iq(1).m) through GitHub issue searches for performance, CPU training, SVM, and related terms. This is the first reported instance of this model configuration exceeding the 60-second CPU epoch threshold, constituting a new public defect."
}
```

## Deterministic failure packet

kind=performance
id=cfe34bfecee92f87
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=clean
cursor=seed:0 cursor:14520 next:14521 composition:114179120289960
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:cpu
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-454-14520/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/temporal/series_sqlite.sqlite")
		.target("target")
		.split(0.8);
	let model = recipe.model()
		.svm().silu().norm(batch).qi(4).nf
		.svm().exp().norm(batch).iq(1).m
		.loss(ce);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(114179120289960)
		.epochs(1)
		.log(all)
		.stop(0.8)
		.fp(32)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=cpu

**Comment by nm-z** [comment]
The public `recipe.train().run()` remains active past 60 seconds on CPU before epoch execution. The earliest failing path is `lower_estimator` for `.svm().exp().qi(5).0`, which runs `fit_svm` and then emits the SVM predictor for `compile_model` -> `compile_native_artifact` through the CPU native compiler. The pool and SQLite input add only a small contraction and memory workload, with no tree candidates or boosting iterations, so the expected path is under 10 seconds; the dominant gap is the SVM predictor preparation/native compilation stall already tracked by #501.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `ae5977f656ab6465`
- Classifier provider: `copilot`
- Classifier model: `gpt-5.6-luna`
- Classifier effort: `auto`

### Classifier decision

```json
{
  "verdict": "comment",
  "issue": 501,
  "title": "CPU training epoch exceeds 60 seconds with SVM-silu-norm-qi(4).nf-svm-exp-norm-iq(1).m model on temporal dataset",
  "body": "The public `recipe.train().run()` remains active past 60 seconds on CPU before epoch execution. The earliest failing path is `lower_estimator` for `.svm().exp().qi(5).0`, which runs `fit_svm` and then emits the SVM predictor for `compile_model` -> `compile_native_artifact` through the CPU native compiler. The pool and SQLite input add only a small contraction and memory workload, with no tree candidates or boosting iterations, so the expected path is under 10 seconds; the dominant gap is the SVM predictor preparation/native compilation stall already tracked by #501.",
  "rationale": "Issue #501 represents the same earliest SVM estimator path on CPU. The current `.pool(2).leak().iq(3).xxs` block, temporal SQLite representation, loss, precision, and optimizer do not change the first failing SVM preparation and native predictor compilation path. The measured 60 seconds is materially slower than the small row/feature workload justifies, so this is additional performance evidence rather than a new cause."
}
```

## Deterministic failure packet

kind=performance
id=ae5977f656ab6465
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=modified
cursor=seed:0 cursor:24266 next:24267 composition:96598494931774
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:cpu
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-9115-24266/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/temporal/series_sqlite.sqlite")
		.target("target");
	let model = recipe.model()
		.pool(2).leak().iq(3).xxs
		.svm().exp().qi(5).0
		.loss(mae);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(96598494931774)
		.epochs(1)
		.log(all)
		.int(4)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=cpu

**Comment by nm-z** [comment]
Measured 60 s on `backend=cpu` for `recipe.train().run()` with `.svm().gelu().iq(4).xs .perc(8).tanh().norm(batch).qi(3).k.l .loss(focal)` on `recipe.data("data/temporal/chronological_splits").target("target").norm(z_score)`. The dataset loads ~12001 rows with one feature (`input`). The SVM predictor (`fit_svm` default 128 iterations, small regularization) plus one width-8 perceptron with tanh, batch normalization, and integer quantization `.iq(4).xs`/`.qi(3).k.l` yields well under 1 MFLOP per epoch and memory traffic under 200 KB; a defensible CPU total (compilation + one epoch) is under 10 s, so 60 s is materially slower. The earliest public failure is `recipe.train().run()` before any epoch completes; the inspected implementation path is the CPU native backend (`compile_model` -> `NativeTape::epoch`) for the quantized batch-normalized SVM-estimator graph. Issue #501 reaches the same earliest operation through the same CPU native-artifact and epoch path (SVM estimator with batch normalization and quantized integer kernels on a temporal dataset). The `.gelu()` vs `.silu()`, `.iq(4).xs` vs `.qi(4).nf`, `.qi(3).k.l` vs `.iq(1).m`, and the extra `.perc(8).tanh()` block are irrelevant configuration differences that do not change the earliest failing operation or implementation path. This packet is additional evidence for the existing SVM-quantization-CPU-stall issue, not a new public defect.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `754374b8d7426e89`
- Classifier provider: `openrouter`
- Classifier model: `thinkingmachines/inkling:free`
- Classifier effort: `default`

### Classifier decision

```json
{
  "verdict": "comment",
  "issue": 501,
  "title": "Performance: CPU training epoch exceeds 60 s for quantized dense-layer plus kmeans predictor with batch normalization on temporal splits",
  "body": "Measured 60 s on `backend=cpu` for `recipe.train().run()` with `.svm().gelu().iq(4).xs .perc(8).tanh().norm(batch).qi(3).k.l .loss(focal)` on `recipe.data(\"data/temporal/chronological_splits\").target(\"target\").norm(z_score)`. The dataset loads ~12001 rows with one feature (`input`). The SVM predictor (`fit_svm` default 128 iterations, small regularization) plus one width-8 perceptron with tanh, batch normalization, and integer quantization `.iq(4).xs`/`.qi(3).k.l` yields well under 1 MFLOP per epoch and memory traffic under 200 KB; a defensible CPU total (compilation + one epoch) is under 10 s, so 60 s is materially slower. The earliest public failure is `recipe.train().run()` before any epoch completes; the inspected implementation path is the CPU native backend (`compile_model` -> `NativeTape::epoch`) for the quantized batch-normalized SVM-estimator graph. Issue #501 reaches the same earliest operation through the same CPU native-artifact and epoch path (SVM estimator with batch normalization and quantized integer kernels on a temporal dataset). The `.gelu()` vs `.silu()`, `.iq(4).xs` vs `.qi(4).nf`, `.qi(3).k.l` vs `.iq(1).m`, and the extra `.perc(8).tanh()` block are irrelevant configuration differences that do not change the earliest failing operation or implementation path. This packet is additional evidence for the existing SVM-quantization-CPU-stall issue, not a new public defect.",
  "rationale": "The reproduction reaches the earliest public failure (`recipe.train().run()` crossing the 60 s slow-runtime threshold) through the CPU native-artifact and epoch path (`compile_model` / `NativeTape::epoch`) with a quantized batch-normalized SVM-estimator graph on `data/temporal/chronological_splits`. Issue #501 covers the same earliest cause: CPU native compilation and epoch overhead for an SVM estimator combined with batch normalization and integer quantization on the same dataset family. The exact quantization profiles (`.iq(4).xs`, `.qi(3).k.l`), activation (`gelu`), and the downstream `.perc(8).tanh()` block differ only in configuration details that execute after or alongside the same failing SVM predictor path, not in the earliest operation or its implementation cause. The measured 60 s gap is dominated by the same CPU backend overhead rather than arithmetic, memory traffic, or loader time, confirming this is a repeated instance of issue #501 rather than a distinct performance defect."
}
```

## Deterministic failure packet

kind=performance
id=754374b8d7426e89
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=modified
cursor=seed:0 cursor:22655 next:22656 composition:48497760879565
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:cpu
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-machine-workload/target/debug/recipe-trials/recipe-trial-6465-22655/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/temporal/chronological_splits")
		.target("target")
		.norm(z_score);
	let model = recipe.model()
		.svm().gelu().iq(4).xs
		.perc(8).tanh().norm(batch).qi(3).k.l
		.loss(focal);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(48497760879565)
		.epochs(1)
		.log(all)
		.stop(0.0)
		.bf(16)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=cpu
